Makefile

```diff
@@ -0,0 +1,39 @@
+SHELL=/bin/bash
+.PHONY: build-ngsa-app create delete
+
+all: create deploy-ngsa-memory
+
+delete:
+	@k3d cluster delete ngsa-app
+
+create: delete
+	@k3d cluster create ngsa-app --registry-use k3d-registry.localhost:5000 --config deploy/k3d.yaml --k3s-arg "--no-deploy=traefik@server:0"
+
+	@kubectl wait node --for condition=ready --all --timeout=60s
+	@sleep 5
+	@kubectl wait pod -A --all --for condition=ready --timeout=60s
+
+	@istioctl install -y --set profile=demo -f deploy/istio-operator.yaml
+
+	@kubectl create namespace ngsa
+
+	@kubectl label namespace ngsa istio-injection=enabled --overwrite
+
+build-ngsa-app:
+	docker build . -t localhost:5000/ngsa-app:local
+	docker push localhost:5000/ngsa-app:local
+
```

Comment:
Do you think we should add a delete-app Makefile target?
It will enable us to delete only the apps without worrying about port conflict and manual deletion.

```Makefile
delete-ngsa-apps:
  -@kubectl delete --ignore-not-found -f deploy/ngsa-memory.yaml
  -@kubectl delete --ignore-not-found -f deploy/ngsa-cosmos.yaml

# Add delete-ngsa-apps as a pre-target
deploy-ngsa-memory: build-ngsa-app delete-ngsa-apps
  ...
  ...

# Add delete-ngsa-apps as a pre-target
deploy-ngsa-cosmos: build-ngsa-app delete-ngsa-apps
  ...
  ...
```

Just a suggestion. Totally your call if you want to include it.

Reply:
added